src/containers/audio-rewards-page/components/SendInputBody.tsx

```diff
@@ -54,49 +61,92 @@ type BalanceError =
   | 'INSUFFICIENT_TRANSFER_AMOUNT'
   | 'EMPTY'
   | 'MALFORMED'
-type AddressError = 'MALFORMED' | 'EMPTY' | 'SEND_TO_SELF'
+  | 'LESS_THAN_MIN'
+
+type AddressError =
+  | 'MALFORMED'
+  | 'EMPTY'
+  | 'SEND_TO_SELF'
+  | 'INVALID_SPL_ADDRESS'
+
+const makeMinAudioError = (num: number | string) =>
+  `You must send at least ${num} $AUDIO`
 
 const balanceErrorMap: { [B in BalanceError]: string } = {
   INSUFFICIENT_BALANCE: messages.insufficientBalance,
   EMPTY: messages.amountRequired,
   MALFORMED: messages.amountMalformed,
-  INSUFFICIENT_TRANSFER_AMOUNT: messages.amountInsufficient
+  INSUFFICIENT_TRANSFER_AMOUNT: messages.amountInsufficient,
+  LESS_THAN_MIN: 'LESS_THAN_MIN' // special case this key to create on the fly
 }
 
 const addressErrorMap: { [A in AddressError]: string } = {
   MALFORMED: messages.addressMalformed,
   EMPTY: messages.addressRequired,
-  SEND_TO_SELF: messages.addressIsSelf
+  SEND_TO_SELF: messages.addressIsSelf,
+  INVALID_SPL_ADDRESS: messages.validSPLAddress
 }
 
 type SendInputBodyProps = {
   currentBalance: BNWei
-  onSend: (balance: BNWei, destinationAddress: WalletAddress) => void
+  onSend: (
+    balance: BNWei,
+    destinationAddress: WalletAddress,
+    chain: Chain
+  ) => void
   wallet: WalletAddress
+  solWallet: WalletAddress
 }
 
-const isValidDestination = (wallet: WalletAddress) => {
+const isValidEthDestination = (wallet: WalletAddress) => {
   const libs = window.audiusLibs
   return libs.web3Manager.web3.utils.isAddress(wallet)
 }
 
+const isValidSolDestination = (wallet: WalletAddress) => {
```

Comment:
this is an interesting one - we should think through whether this is sufficient, or if we should actually check for the existence of a Token Account at this address first. Just sending to a regular Solana address will fail since it's not a token account created for our mint. would be good to see how Sol wallets (phantom) handle this

Reply:
I figured it would be fine to check for valid solana address and just error on sending. 
This is a good point, can check what the behavior of phantom is.

Reply:
caught up on slack - system sol account will work since we'll derive the token account